src/writer.rs

```diff
@@ -394,4 +417,39 @@ mod indentation {
                 .as_ref()
         );
     }
+    #[test]
+    fn node_writer_empty() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_node(b"empty")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_empty()
+            .expect("failure");
+
+        assert_eq!(
+            std::str::from_utf8(&buffer).unwrap(),
+            r#"<empty attr1="value1" attr2="value2"/>"#
+        );
+    }
+
+    #[test]
+    fn node_writer_text() {
+        let mut buffer = Vec::new();
+        let mut writer = Writer::new_with_indent(&mut buffer, b' ', 4);
+
+        writer
+            .create_node(b"paired")
+            .with_attribute(("attr1", "value1"))
+            .with_attribute(("attr2", "value2"))
+            .write_text(BytesText::from_plain_str("text"))
```

Comment:
Should a ```write_cdata``` be added?

Reply:
Yes ... ideally all types of events

Reply:
All events, or all events of the type that would make sense to be nested in a single pair of tags?  I don't use XML outside of the one project I'm working on so I don't know - but based on what I could find, I thought that Text, CData, and maybe PI are the only ones you'd typically see in this context.  You probably wouldn't want to write random Start, End, Decl, EOF, etc.

 If you want to allow *any* type of event, we could just add a ```.write_event()```, but in my mind this is more of a simple API for the standard case, and if you want to reproduce weird / invalid XML perfectly then you would just use the existing API.

Reply:
decl probably doesn't make any sense indeed